packages/venia-ui/lib/components/RichContent/PageBuilder/ContentTypes/ButtonItem/configAggregator.js

```diff
@@ -0,0 +1,33 @@
+import { getAdvanced, getCssClasses } from '../../utils';
+
+export default node => {
```

Comment:
So this code:

1. Sets up a base object
2. Calculates some things
3. Mutates base object
4. Returns

I personally tend to find things easier to read when the object is setup at once, so I have a clear picture of the shape (since there are no types here). Consider pulling some things out into well-named variables and creating the object at once. Example:

```js
export default node => {
    const { firstChild, style, textContent } = node;

    const advancedProps = getAdvanced(firstChild);
    const classes = getCssClasses(node);
    const isSecondary = firstChild.classList.containes('pagebuilder-button-secondary');
    const isBtnLink = firstChild.classList.containes('pagebuilder-button-link');
    const isLinkTag = firstChild.nodeName === 'A';

    return {
        display: style.display,
        text: textContent,
        textAlign: style.textAlign,
        openInNewTab: firstChild.target === '_blank',
        buttonType: isSecondary ? 'secondary' : isBtnLink ? 'link': 'primary',
        ...advancedProps,
        ...classes,
        ...(isLinkTag && {
            link: firstChild.href,
            linkType: firstChild.getAttribute('data-link-type'),
        })
    };
};
```

Reply:
Nice tip! We'll keep this in mind when writing other configAggregators. This will be resolved as part of the "Buttons" story: https://jira.corp.magento.com/browse/PB-12